pkg/vault/test/foundry/utils/BaseMedusaTest.sol

```diff
@@ -0,0 +1,223 @@
+    IVaultMock internal vault;
+    IVaultExtension internal vaultExtension;
+    IVaultAdmin internal vaultAdmin;
+    RouterMock internal router;
+    BatchRouterMock internal batchRouter;
+    CompositeLiquidityRouterMock internal compositeLiquidityRouter;
+    BasicAuthorizerMock internal authorizer;
+    ProtocolFeeControllerMock internal feeController;
+    PoolFactoryMock internal factoryMock;
+
+    IBasePool internal pool;
+    uint256 internal poolCreationNonce;
+
+    uint256 internal constant DEFAULT_USER_BALANCE = 1e18 * 1e18;
+
+    // Set alice,bob and lp to addresses of medusa.json "senderAddresses" property
+    address internal alice = address(0x10000);
+    address internal bob = address(0x20000);
+    address internal lp = address(0x30000);
+
+    ERC20TestToken internal dai;
+    ERC20TestToken internal usdc;
+    WETHTestToken internal weth;
+
+    constructor() {
+        dai = _createERC20TestToken("DAI", "DAI", 18);
+        usdc = _createERC20TestToken("USDC", "USDC", 18);
+        weth = new WETHTestToken();
+
+        // The only function used by _mintTokenToUsers is mint, which has the same signature as ERC20TestToken. So,
+        // cast weth as ERC20TestToken to use the same funtion.
+        _mintTokenToUsers(ERC20TestToken(address(weth)));
+
+        DeployPermit2 _deployPermit2 = new DeployPermit2();
+        permit2 = IPermit2(_deployPermit2.deployPermit2());
+
+        _deployVaultMock(0, 0);
+
+        router = new RouterMock(IVault(address(vault)), weth, permit2);
+        batchRouter = new BatchRouterMock(IVault(address(vault)), weth, permit2);
+        compositeLiquidityRouter = new CompositeLiquidityRouterMock(IVault(address(vault)), weth, permit2);
+
+        _setPermissionsForUsersAndTokens();
+
+        (IERC20[] memory tokens, uint256[] memory initialBalances) = getTokensAndInitialBalances();
+        pool = IBasePool(createPool(tokens, initialBalances));
+
+        _allowBptTransfers();
+    }
+
+    function createPool(IERC20[] memory tokens, uint256[] memory initialBalances) internal virtual returns (address) {
+        address newPool = factoryMock.createPool("ERC20 Pool", "ERC20POOL");
+
+        // No hooks contract.
+        factoryMock.registerTestPool(newPool, vault.buildTokenConfig(tokens), address(0), lp);
+
+        // Initialize liquidity of new pool.
+        medusa.prank(lp);
+        router.initialize(address(newPool), tokens, initialBalances, 0, false, bytes(""));
+
+        return newPool;
+    }
+
+    function getTokensAndInitialBalances()
+        internal
+        virtual
+        returns (IERC20[] memory tokens, uint256[] memory initialBalances)
+    {
+        tokens = new IERC20[](3);
+        tokens[0] = dai;
+        tokens[1] = usdc;
+        tokens[2] = weth;
+        tokens = InputHelpers.sortTokens(tokens);
+
+        initialBalances = new uint256[](3);
+        initialBalances[0] = 10e18;
+        initialBalances[1] = 20e18;
+        initialBalances[2] = 30e18;
```

Comment:
Any reason to use these?

Reply:
Not really, it was just a way to strt the tests with an unbalanced pool, but we can start with balanced pools, no problem. I just changed it.